esmvaltool/diag_scripts/cvdp/cvdp_wrapper.py

```diff
@@ -127,11 +134,180 @@ def _nco_available():
+        'za': "mean",
+    }
+    ans = _get_info(filename, stat)
+    return ans if ans is not None else 'other'
+
+
+def _get_season(filename):
+    season = {
+        'ann': "Annual",
+        'djf': "DJF",
+        'mam': "MAM",
+        'jja': "JJA",
+        'son': "SON",
+    }
+    return _get_info(filename, season)
+
+
+def _get_long_name(filename):
+    variable = {
+        'pr': "Precipitation",
+        'tas': "Surface temperature",
+        'psl': "Sea level pressure",
+        'sst': "Sea surface temperature",
+    }
+    return _get_info(filename, variable)
+
+
+def _get_mode(filename):
+    mode = {
+        'iod': "iod",
+        'ipo': "ipo",
+        'nam': "nam",
+        'nao': "nao",
+        'lanina': "La nina",
+        'nino12': "El nino 12",
+        'nino3': "El nino 3",
+        'nino34': "El nino 34",
+        'nino4': "El nino 4",
+        'npi': "npi",
+        'npo': "npo",
+        'pdo': "pdo",
+        'pna': "pna",
+        'psa1': "psa1",
+        'psa2': "psa2",
+        'sam': "sam",
+        'socn': "socn",
+        'tio': "tio",
+        'tna': "tna",
+        'tsa': "tsa",
+    }
+    return _get_info(filename, mode)
+
+
+def _get_info(filename, dictionary):
+    intersection = list(
+        set(os.path.basename(filename).split('.')).intersection(
+            dictionary.keys()))
+    if len(intersection) != 1:
+        return None
+    return dictionary[intersection[0]]
+
+
+def _get_global_ancesters(cfg):
+    input_data = cfg['input_data'].values()
+    selection = select_metadata(input_data, project='CMIP5')
+    grouped_selection = group_metadata(selection, 'dataset')
+    ancester = []
+    for _, attributes in grouped_selection.items():
+        ancester += [item['filename'] for item in attributes]
+    return ancester
+
+
+def set_provenance(cfg):
+    """Add provenance to all image files that the cvdp package creates."""
+
+    def _get_provenace_record(filename, ancesters):
+        return {
+            'caption': _get_caption(filename),
+            'statistics': [_get_stat(filename)],
+            'domain': 'global',
+            'plot_type': _get_plot_type(filename),
+            'plot_file': filename,
+            'authors': [
+                'phil_ad',
+            ],
+            'references': [
+                'acknow_project',
+            ],
+            'ancesters': ancesters,
+        }
+
+    ancesters = _get_global_ancesters(cfg)
+    logger.info("Path to work_dir: %s", cfg['work_dir'])
+    for root, _, files in os.walk(cfg['work_dir']):
+        for datei in files:
+            path = os.path.join(root, datei)
+            if _is_png(path):
+                logger.info("Name of file: %s", path)
+                provenance_record = _get_provenace_record(path, ancesters)
+                logger.info("Recording provenance of %s:\n%s", path,
+                            provenance_record)
+                with ProvenanceLogger(cfg) as provenance_logger:
```

Comment:
this line can be moved outside the for loops, i.e. line 289

Reply:
done